### PR TITLE
Vendor model_builder from pymc-experimental

### PR DESCRIPTION
**Pull request by maresb**
This way we don't need an awkward dependency on pymc-experimental.

----
:books: Documentation preview :books:: https://pymc-marketing--339.org.readthedocs.build/en/339/

### CODE REVIEW

**Review by twiecki — APPROVED** [review]

File: pymc_marketing/vendored/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Vendored files
+
+Vendored files should never be edited in this repository.
+Instead, edit the originals and copy them here.
```
[the remark this answers is not in the document]
Comment (michaelraczycki) [reply pymc_marketing/vendored/README.md:4]: Same as Ricardo, I understood that now we'd move MB completely to marketing

[the remark this answers is not in the document]
Comment (maresb) [reply pymc_marketing/vendored/README.md:4]: Whatever you decide. Anything is possible. :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (maresb) [reply pymc_marketing/vendored/README.md]: Moved into the main package

**Comment by maresb** [comment]
Hmm, now there are a bunch of mypy errors that need to be resolved.

Feel free to replace this PR if that's easier.

**Comment by michaelraczycki** [comment]
This is something I've been warning about later, that pymc-experimental and pymc-marketing is not mutually compatible when it comes to linters. This needs to be addressed if pymc-experimental is meant to stay as the staging area for new features for other packages

**Comment by maresb** [comment]
Anyone else actively fixing the mypy errors or shall I take a quick stab at it?

**Comment by ricardoV94** [comment]
> Anyone else actively fixing the mypy errors or shall I take a quick stab at it?

Feel free to take a stab!

**Comment by maresb** [comment]
I made a tiny bit of progress, but there's much more to go. I'm not working on it at the moment if anyone else wants to move this forward.

**Comment by maresb** [comment]
The current tests for ModelBuilder don't quite reach the target, but I think this may be mergeable now.

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
CC @michaelraczycki any issue you spot last minute?

**Comment by michaelraczycki** [comment]
it's a good starting point, we can improve later the test coverage etc.

**Pull request merged**